compaction/compaction_manager.cc

```diff
@@ -762,6 +764,7 @@ future<> compaction_manager::postponed_compactions_reevaluation() {
                 auto s = t->schema();
                 cmlog.debug("resubmitting postponed compaction for table {}.{} [{}]", s->ks_name(), s->cf_name(), fmt::ptr(t));
                 submit(*t);
+                co_await coroutine::maybe_yield();
```

Comment:
Could the remaining tasks in postponed become invalid while we yield? Attempted to be removed from _postponed, but since they were moved at the top of the loop, missed.

btw, when grabbing data, better to use std::exchange() since std::move() doesn't guarantee a copy will not happen (depends on type and constness).

Reply:
they can become invalid, but once we try to run them, manager detects it's no longer registered and throws  std::out_of_range during submission.

I can also do this to make sure table_state deregistration waits for reevaluation holding table_state ref to stop doing so:

```
diff --git a/compaction/compaction_manager.cc b/compaction/compaction_manager.cc
index 6ed4f1cb05..d81d49b104 100644
--- a/compaction/compaction_manager.cc
+++ b/compaction/compaction_manager.cc
@@ -759,6 +759,9 @@ future<> compaction_manager::postponed_compactions_reevaluation() {
             co_return;
         }
         auto postponed = std::move(_postponed);
+        auto holders = boost::copy_range<std::vector<seastar::gate::holder>>(postponed | boost::adaptors::transformed([this] (compaction::table_state* t) {
+            return get_compaction_state(t).gate.hold();
+        }));
         try {
             for (auto& t : postponed) {
                 auto s = t->schema();
```

which makes sense.